flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -814,6 +814,7 @@ private void connect(Integer headOfChain, StreamEdge edge) {
         }
         // set strategy name so that web interface can show it.
         jobEdge.setShipStrategyName(partitioner.toString());
+        jobEdge.setBroadcast(partitioner.isBroadcast());
```

Comment:
I think `setBroadcast` is also needed for dataset jobs, i.e.  in `JobGraphGenerator`.